test/abstract_client.js

```diff
@@ -419,6 +419,25 @@ module.exports = function (server, config) {
       });
     });
 
+    it('should emit a publish event', function (done) {
+      var client = connect(),
+        payload = 'test_payload',
+        topic = 'test_topic';
+
+      client.once('publish', function (topic, message, packet) {
+        topic.should.equal(topic);
```

Comment:
At this point it is not 100% clear to which `topic` is referenced, you can use `packet.topic.should.equal(topic)` or rename `topic = 'test_topic'` to `testTopic = 'test_topic'`